### ISSUE TITLE
Table index 0 causes infinite loop

### ISSUE THREAD

**Issue by xconstruct** [yuin/gopher-lua#81]
The following code causes an infinite loop for me on master/Go 1.7/Linux:
```lua
local tbl = {
        [-1] = "a",
        [0] = "b",
        [1] = "c",
}
for k, v in pairs(tbl) do
        print(k, v)
end
```
Output:
```
-1      a
0       b
1       c
-1      a
0       b
1       c
... and so on
```

It only happens with numeric zero (not string). I know Lua indexing starts with 1 instead of 0. But in some calculations a zero may be stored in tables, and this loop does not happen in reference Lua 5.1.

Minimal test case:
```lua
local tbl = { [0] = true }
assert(next(tbl, 0) == nil) -- should be nil, but is 0
```

**Comment by yuin** [comment]
Thanks for reporting this.
I've fixed this bug in be46d3177260d2a4042c3dc86528816713b37ffb .

**Comment by xconstruct** [comment]
Thank you for your extremely fast response and fix!
